src/ORM/DataQuery.php

```diff
@@ -1142,16 +1203,51 @@ public function addSelectFromTable($table, $fields)
 
     /**
      * Query the given field column from the database and return as an array.
+     * querying DB columns of related tables is supported but you need to make sure that the related table
+     * is already available in join
+     *
+     * @see DataList::applyRelation()
+     *
+     * example use:
+     *
+     * <code>
+     *  column("<table_name>"."<column_name>")
+     * </code>
      *
      * @param string $field See {@link expressionForField()}.
      * @return array List of column values for the specified column
+     * @throws InvalidArgumentException
      */
     public function column($field = 'ID')
     {
         $fieldExpression = $this->expressionForField($field);
-        $query = $this->getFinalisedQuery(array($field));
+        $query = $this->getFinalisedQuery([$field]);
         $originalSelect = $query->getSelect();
-        $query->setSelect(array());
+        $query->setSelect([]);
+
+        // field wasn't recognised as a valid field from the table class hierarchy
+        // check if the field is in format "<table_name>"."<column_name>"
+        // if that's the case we may want to query related table
+        if (!$fieldExpression) {
+            $exception = new InvalidArgumentException('Invalid column name ' . $field);
+
+            // standard column - nothing to process here
+            if (strpos($field, '.') === false) {
+                throw $exception;
+            }
+
+            $fieldData = explode('.', $field);
+            $tablePrefix = str_replace('"', '', $fieldData[0]);
+
+            // check if related table is available
+            if (!$query->isJoinedTo($tablePrefix)) {
+                throw $exception;
+            }
+
+            $fieldExpression = $field;
+            $field = null;
+        }
```

Comment:
I think I'd prefer to see this in its own method, maybe something like `validateFieldExpression()`